packages/core/src/Stores/pushwoosh-store.js

```diff
@@ -19,6 +19,18 @@ export default class PushwooshStore extends BaseStore {
      */
     @action.bound
     init = () => {
+        let reaction_id;
+        if (!this.root_store.common.is_network_online) {
+            reaction_id = when(
+                () => this.root_store.common.is_network_online,
+                () => {
+                    this.init();
+                }
+            );
+            return;
+        }
+        // canceling the reaction once is_network_online becomes true for the first time:
+        reaction_id?.cancel();
```

Comment:
this can be replaced by using when with await
```suggestion
        await when(()=> this.root_store.common.is_network_online);
```

Reply:
thank you so much! 😄 And I've also figured out that I don't need to cancel the `when` reaction because `when` will cancel itself after the first run (when is_network_online becomes true for the first time)